velox/functions/prestosql/QDigestFunctions.h

```diff
@@ -158,4 +158,100 @@ struct QuantileAtValueFunction {
   }
 };
 
+template <typename TExec>
+struct ScaleQDigestDoubleFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<SimpleQDigest<double>>& result,
+      const arg_type<SimpleQDigest<double>>& input,
+      const arg_type<double>& scaleFactor) {
+    std::allocator<double> allocator;
+    qdigest::QuantileDigest<double, std::allocator<double>> digest(
+        allocator, input.data());
+    digest.scale(scaleFactor);
+    int64_t size = digest.serializedByteSize();
+    result.resize(size);
+    digest.serialize(result.data());
+  }
+};
+
+template <typename TExec>
+struct ScaleQDigestBigintFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<SimpleQDigest<int64_t>>& result,
+      const arg_type<SimpleQDigest<int64_t>>& input,
+      const arg_type<double>& scaleFactor) {
+    std::allocator<int64_t> allocator;
+    qdigest::QuantileDigest<int64_t, std::allocator<int64_t>> digest(
+        allocator, input.data());
+    digest.scale(scaleFactor);
+    int64_t size = digest.serializedByteSize();
+    result.resize(size);
+    digest.serialize(result.data());
+  }
+};
+
+template <typename TExec>
+struct ScaleQDigestRealFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(TExec);
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<SimpleQDigest<float>>& result,
+      const arg_type<SimpleQDigest<float>>& input,
+      const arg_type<double>& scaleFactor) {
+    std::allocator<float> allocator;
+    qdigest::QuantileDigest<float, std::allocator<float>> digest(
+        allocator, input.data());
+    digest.scale(scaleFactor);
+    int64_t size = digest.serializedByteSize();
+    result.resize(size);
+    digest.serialize(result.data());
+  }
+};
```

Comment:
@natashasehgal  Do you still need these three separate classes since you're combining them into the ScaleQDigestFunction class below? The rest of this PR looks good to me.

Reply:
Thanks for the code reference. I think the issue is because SimpleQDigest<T> is a complex type. If I use T and TInput, and use TInput& instead of arg_type I get 

E.g. 
error: no member named 'udf' in 'facebook::velox::functions::ScaleQDigestFunction<double>'
   47 |   using funcClass = typename Func::template udf<exec::VectorExec>;
      |                              ~~~~~~         ^

Reply:
Done!

Btw, on Meta using just template T worked but on Github there was a complaint about using C++ 20 functionality. So I moved to a template with T, ResultT, InputT

In file included from /__w/velox/velox/velox/functions/prestosql/registration/QDigestFunctionsRegistration.cpp:17:
/__w/velox/velox/./velox/functions/prestosql/QDigestFunctions.h:165:12: error: use of 'auto' in parameter declaration only available with '-std=c++20' or '-fconcepts' [-Werror]
  165 |   callImpl(auto& result, const auto& input, auto scaleFactor) {